sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -551,28 +629,452 @@ class ParquetPartitionReader(
+          addPartitionValues(maybeBatch, partValues)
+        } finally {
+          table.foreach(_.close())
+        }
+      }
+    }
+  }
+
+  private def readToTable(
+      currentChunkedBlocks: Seq[(Path, BlockMetaData)],
+      currentClippedSchema: MessageType): Option[Table] = {
+    if (currentChunkedBlocks.isEmpty) {
+      return None
+    }
+    val (dataBuffer, dataSize) = readPartFiles(currentChunkedBlocks, currentClippedSchema)
+    try {
+      if (dataSize == 0) {
+        None
+      } else {
+        if (debugDumpPrefix != null) {
+          dumpParquetData(dataBuffer, dataSize, splits)
+        }
+        val parseOpts = ParquetOptions.builder()
+          .withTimeUnit(DType.TIMESTAMP_MICROSECONDS)
+          .includeColumn(readDataSchema.fieldNames:_*).build()
+
+        // about to start using the GPU
+        GpuSemaphore.acquireIfNecessary(TaskContext.get())
+
+        val table = withResource(new NvtxWithMetrics("Parquet decode", NvtxColor.DARK_GREEN,
+            metrics(GPU_DECODE_TIME))) { _ =>
+          Table.readParquet(parseOpts, dataBuffer, 0, dataSize)
+        }
+        if (!isCorrectedRebaseMode) {
+          (0 until table.getNumberOfColumns).foreach { i =>
+            if (RebaseHelper.isDateTimeRebaseNeededRead(table.getColumn(i))) {
+              throw RebaseHelper.newRebaseExceptionInRead("Parquet")
+            }
+          }
+        }
+        maxDeviceMemory = max(GpuColumnVector.getTotalDeviceMemoryUsed(table), maxDeviceMemory)
+        if (readDataSchema.length < table.getNumberOfColumns) {
+          table.close()
+          throw new QueryExecutionException(s"Expected ${readDataSchema.length} columns " +
+            s"but read ${table.getNumberOfColumns} from $currentChunkedBlocks")
+        }
+        metrics(NUM_OUTPUT_BATCHES) += 1
+        Some(evolveSchemaIfNeededAndClose(table, splits.mkString(","), currentClippedSchema))
+      }
+    } finally {
+      dataBuffer.close()
+    }
+  }
+
+  private def populateCurrentBlockChunk():
+    (MessageType, InternalRow, Seq[(Path, BlockMetaData)]) = {
+
+    val currentChunk = new ArrayBuffer[(Path, BlockMetaData)]
+    var numRows: Long = 0
+    var numBytes: Long = 0
+    var numParquetBytes: Long = 0
+    var currentFile: Path = null
+    var currentPartitionValues: InternalRow = null
+    var currentClippedSchema: MessageType = null
+
+    @tailrec
+    def readNextBatch(): Unit = {
+      if (blockIterator.hasNext) {
+        if (currentFile == null) {
+          currentFile = blockIterator.head.filePath
+          currentPartitionValues = blockIterator.head.partValues
+          currentClippedSchema = blockIterator.head.schema
+        }
+        if (currentFile != blockIterator.head.filePath) {
+          // check to see if partitionValues different, then have to split it
+          if (blockIterator.head.partValues != currentPartitionValues) {
+            logWarning(s"Partition values for the next file ${blockIterator.head.filePath}" +
```

Comment:
Are these warnings here because the partition factory should never have generated a single partition with these attributes, or will this naturally happen with some datasets?